src/EnergyPlus/PVWatts.cc

```diff
@@ -466,21 +466,20 @@ namespace PVWatts {
 
     PVWattsGenerator &GetOrCreatePVWattsGenerator(EnergyPlusData &state, std::string const &GeneratorName)
     {
-        // Find the generator, and create a new one if it hasn't been loaded yet.
-        int ObjNum =
-            state.dataInputProcessing->inputProcessor->getObjectItemNum(state, "Generator:PVWatts", UtilityRoutines::MakeUPPERCase(GeneratorName));
-        assert(ObjNum >= 0);
-        if (ObjNum == 0) {
-            ShowFatalError(state, "Cannot find Generator:PVWatts " + GeneratorName);
-        }
-
         auto &PVWattsGenerators(state.dataPVWatts->PVWattsGenerators);
 
-        auto it = PVWattsGenerators.find(ObjNum);
+        auto it = PVWattsGenerators.find(GeneratorName);
         if (it == PVWattsGenerators.end()) {
             // It's not in the map, add it.
-            PVWattsGenerators.insert(std::make_pair(ObjNum, PVWattsGenerator::createFromIdfObj(state, ObjNum)));
-            PVWattsGenerator &pvw(PVWattsGenerators.find(ObjNum)->second);
+            int ObjNum =
+                state.dataInputProcessing->inputProcessor->getObjectItemNum(state, "Generator:PVWatts", UtilityRoutines::MakeUPPERCase(GeneratorName));
+            assert(ObjNum >= 0);
+            if (ObjNum == 0) {
+                ShowFatalError(state, "Cannot find Generator:PVWatts " + GeneratorName);
+            }
+
+            PVWattsGenerators.insert(std::make_pair(GeneratorName, PVWattsGenerator::createFromIdfObj(state, ObjNum)));
+            PVWattsGenerator &pvw(PVWattsGenerators.find(GeneratorName)->second);
             pvw.setupOutputVariables(state);
             return pvw;
```

Comment:
This lookup of the PVWatts object in the list was adding some time mostly because we called the input processor every time around to get the `ObjNum` from the object name. Changed the `std::map` that holds those to have the names as keys instead of `ObjNum` and it shaved some time off.